src/compile/compile.ts

```diff
@@ -56,11 +56,18 @@ function assemble(model: Model, topLevelProperties: TopLevelProperties) {
   // Config with Vega-Lite only config removed.
   const vgConfig = model.config ? stripConfig(model.config) : undefined;
 
+  // remove autoResize as it has to be put under autosize
+  const topLevelProps = extend({}, topLevelProperties);
+  delete topLevelProps.autoResize;
+
   const output = {
     $schema: 'http://vega.github.io/schema/vega/v3.0.json',
     ...(model.description ? {description: model.description} : {}),
-    autosize: 'pad', // By using Vega layout, we don't support custom autosize
-    ...topLevelProperties,
+    autosize: {
```

Comment:
Wonder if you think
```
autosize: topLevelProperties.autoResize ? {type: 'pad', resize: true} : 'pad'
```
is a bit nicer, and won't require changing all existing specs.